### PR TITLE
Feature 34 pages under devices

### LINKED ISSUE #34: Pages under devices
Exhibtionpages need to have device id

### PR DESCRIPTION
**Pull request by jarinykanen**
Resolves #34

### CODE REVIEW

File: src/main/kotlin/fi/metatavu/muisti/persistence/dao/ExhibitionPageDAO.kt [hunk 1]
```diff
@@ -48,17 +47,26 @@ class ExhibitionPageDAO() : AbstractDAO<ExhibitionPage>() {
...
+    fun listByExhibition(exhibition: Exhibition, exhibitionDevice : ExhibitionDevice?): List<ExhibitionPage> {
         val entityManager = getEntityManager()
         val criteriaBuilder = entityManager.criteriaBuilder
         val criteria: CriteriaQuery<ExhibitionPage> = criteriaBuilder.createQuery(ExhibitionPage::class.java)
         val root: Root<ExhibitionPage> = criteria.from(ExhibitionPage::class.java)
+
+        val restrictions = ArrayList<Predicate>()
+        restrictions.add(criteriaBuilder.equal(root.get(ExhibitionPage_.exhibition), exhibition))
+
+        if(exhibitionDevice != null){
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/muisti/persistence/dao/ExhibitionPageDAO.kt]:
```suggestion
        if (exhibitionDevice != null) {
```

File: src/main/kotlin/fi/metatavu/muisti/persistence/dao/ExhibitionPageDAO.kt [hunk 2]
```diff
@@ -48,17 +47,26 @@ class ExhibitionPageDAO() : AbstractDAO<ExhibitionPage>() {
      * Lists ExhibitionPages by exhibition
      *
      * @param exhibition exhibition
+     * @param exhibitionDevice filter by exhibition device. Ignored if null is passed
      * @return List of ExhibitionPages
      */
-    fun listByExhibition(exhibition: Exhibition): List<ExhibitionPage> {
+    fun listByExhibition(exhibition: Exhibition, exhibitionDevice : ExhibitionDevice?): List<ExhibitionPage> {
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/muisti/persistence/dao/ExhibitionPageDAO.kt]: This method no longer lists just by exbition, so more appropriate name would be just lsit
```suggestion
    fun list(exhibition: Exhibition, exhibitionDevice : ExhibitionDevice?): List<ExhibitionPage> {
```

File: src/main/kotlin/fi/metatavu/muisti/persistence/dao/ExhibitionPageDAO.kt [hunk 3]
```diff
@@ -91,6 +99,19 @@ class ExhibitionPageDAO() : AbstractDAO<ExhibitionPage>() {
...
     }
 
+    /**
+     * Updates device
+     *
+     * @param device device
+     * @param lastModifierId last modifier's id
+     * @return updated exhibitionPage
+     */
+    fun updateDevice(exhibitionPage: ExhibitionPage, device: ExhibitionDevice, lastModifierId: UUID): ExhibitionPage {
```
Comment (MrWalrus3451) [inline src/main/kotlin/fi/metatavu/muisti/persistence/dao/ExhibitionPageDAO.kt:109]: Missing param from comment.

File: src/test/kotlin/fi/metatavu/muisti/api/test/functional/builder/impl/ExhibitionPageTestBuilderResource.kt [hunk 4]
```diff
@@ -66,8 +67,8 @@ class ExhibitionPageTestBuilderResource(testBuilder: AbstractTestBuilder<ApiClie
      * @param exhibitionId exhibition id
      * @return exhibition Pages
      */
-    fun listExhibitionPages(exhibitionId: UUID): Array<ExhibitionPage> {
-        return api.listExhibitionPages(exhibitionId)
+    fun listExhibitionPages(exhibitionId: UUID, exhibitionDeviceId: UUID): Array<ExhibitionPage> {
+        return api.listExhibitionPages(exhibitionId, exhibitionDeviceId)
```
Comment (MrWalrus3451) [inline src/test/kotlin/fi/metatavu/muisti/api/test/functional/builder/impl/ExhibitionPageTestBuilderResource.kt:71]: Missing param exhibition deviceid from comment.

File: src/test/kotlin/fi/metatavu/muisti/api/test/functional/builder/impl/ExhibitionPageTestBuilderResource.kt [hunk 5]
```diff
@@ -115,8 +116,8 @@ class ExhibitionPageTestBuilderResource(testBuilder: AbstractTestBuilder<ApiClie
      * @param expected expected count
      * @param exhibitionId exhibition id
      */
-    fun assertCount(expected: Int, exhibitionId: UUID) {
-        assertEquals(expected, api.listExhibitionPages(exhibitionId).size)
+    fun assertCount(expected: Int, exhibitionId: UUID, exhibitionDeviceId: UUID) {
```
Comment (MrWalrus3451) [inline src/test/kotlin/fi/metatavu/muisti/api/test/functional/builder/impl/ExhibitionPageTestBuilderResource.kt:119]: Missing param exhibitionDeviceId  from comment.

File: src/test/kotlin/fi/metatavu/muisti/api/test/functional/builder/impl/ExhibitionPageTestBuilderResource.kt [hunk 6]
```diff
@@ -152,9 +153,9 @@ class ExhibitionPageTestBuilderResource(testBuilder: AbstractTestBuilder<ApiClie
      * @param expectedStatus expected status
      * @param exhibitionId exhibition id
      */
-    fun assertListFail(expectedStatus: Int, exhibitionId: UUID) {
+    fun assertListFail(expectedStatus: Int, exhibitionId: UUID, exhibitionDeviceId: UUID) {
```
Comment (MrWalrus3451) [inline src/test/kotlin/fi/metatavu/muisti/api/test/functional/builder/impl/ExhibitionPageTestBuilderResource.kt:156]: MIssing param exhibitionDeviceId from comment.

**Pull request merged**